packages/react-composites/src/composites/ChatComposite/adapter/AzureCommunicationChatAdapter.ts

```diff
@@ -393,9 +397,35 @@ export class AzureCommunicationChatAdapter implements ChatAdapter {
     this.emitter.off(event, listener);
   }
 
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private executeWithThreadAccess(callback: () => any): any {
+    if (!this.hasAcceccToThread) {
+      console.error('Chat Adapter does not have access to the thread.');
+    } else {
+      return callback();
+    }
+  }
+
+  private checkIsUserJoinedThread(): void {
+    const check = (): void => {
+      const threads = this.chatClient.listChatThreads({ maxPageSize: 100 }).byPage().next();
+      threads.then((threads) => {
+        if (threads.value.filter((thread: ChatThreadItem) => thread.id === this.chatThreadClient.threadId).length > 0) {
+          this.hasAcceccToThread = true;
+        } else {
+          setTimeout(check, 500);
```

Comment:
don't use polling, instead react to events. we know when the call connects, at that point trigger the chat join